src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentBindLoweringPass.cs

```diff
@@ -340,7 +342,30 @@ private IntermediateNode[] RewriteUsage(IntermediateNode parent, TagHelperProper
                     changeNode.Children[0].Children.Add(changeExpressionTokens[i]);
                 }
 
-                return new[] { valueNode, changeNode };
+                ComponentAttributeIntermediateNode expressionNode = null;
+                if (expressionAttribute != null)
+                {
+                    expressionNode = new ComponentAttributeIntermediateNode(node)
+                    {
+                        AttributeName = expressionAttributeName,
+                        BoundAttribute = expressionAttribute,
+                        PropertyName = expressionAttribute.GetPropertyName(),
+                        TagHelper = node.TagHelper,
+                        TypeName = expressionAttribute.IsWeaklyTyped() ? null : expressionAttribute.TypeName,
+                    };
+
+                    expressionNode.Children.Clear();
+                    expressionNode.Children.Add(new CSharpExpressionIntermediateNode());
+                    expressionNode.Children[0].Children.Add(new IntermediateToken()
+                    {
+                        Content = $"() => {original.Content}",
+                        Kind = TokenKind.CSharp
+                    });
+                }
```

Comment:
Note to @ajaybhargavb - `bind` is an interesting case for debuggability. 

Basically we expect bind to get+set + now create an expression for an LValue. We just make the *get* debuggable. Then if you get a compile error at least you get a reasonable error message and it goes to source correctly.